src/wasm/wasm-validator.cpp

```diff
@@ -199,6 +199,22 @@ struct ValidationInfo {
     fail(text, curr, func);
     return false;
   }
+
+  // Type 'left' should be 'unreachable' or a subtype of 'right'.
+  bool shouldBeSubTypeOrFirstIsUnreachable(Type left,
+                                           Type right,
+                                           Expression* curr,
+                                           const char* text,
+                                           Function* func = nullptr) {
```

Comment:
You can probably just use `shouldBeSubType` for this because `unreachable` is a subtype of everything.

Reply:
Oh interesting. No wonder we didn't have a `shouldBeSubTypeOrFirstIsUnreachable` in the first place. 😅 Updated.